src/emotion/theme.tsx

```diff
@@ -0,0 +1,135 @@
+import { DeepPartial, Dictionary, Omit } from '@stoplight/types';
+import merge = require('lodash/merge');
+import * as React from 'react';
+
+export type BaseTheme = 'dark' | 'light';
+
+/**
+ * Every app should export the result of this function once, and re-use these values.
+ *
+ * This enables typings for zones.
+ */
+export function createThemedModule<S extends string>() {
+  return {
+    ThemeProvider: ThemeProvider as React.SFC<IThemeProvider<S>>,
+    ThemeZone: ThemeZone as IThemeZone<S>,
+    useTheme,
+    useThemeZones: useThemeZones as () => ThemeZones<S>,
+  };
+}
+
+/**
+ * Theme
+ */
+
+interface ITheme {
+  base: BaseTheme;
+
+  canvas: {
+    fg: string;
+    bg: string;
```

Comment:
How to we feel about being more explicit here? So instead of `bg` and `fg` we change to `background` and `foreground`, makes it more clear as an outside user looking to what they mean

Reply:
Medium - what do others think?

If we're not nesting, then things like `selectedItemBg` would become pretty long: `selectedItemBackground`.

Reply:
I'm good either way as long as it auto-completes. In studio we should have a visual layer on top like with linting.

Reply:
I'm good either way as long as it auto-completes. In studio we should have a visual layer on top like with linting.

Reply:
Toss up to me - I'm going to leave as is, think it's fine.